internal/controller/dragonfly_controller.go

```diff
@@ -107,8 +228,31 @@ func (r *DragonflyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 
 		log.Info("Updated resources for object")
-	}
 
+		// perform a rollout only if the pod spec has changed
+		var statefulSet appsv1.StatefulSet
+		if err := r.Get(ctx, client.ObjectKey{Namespace: df.Namespace, Name: df.Name}, &statefulSet); err != nil {
+			log.Error(err, "could not get statefulset")
+			return ctrl.Result{}, err
+		}
+
+		if statefulSet.Status.UpdatedReplicas != statefulSet.Status.Replicas {
+			log.Info("Pod spec has changed, performing a rollout")
+			r.EventRecorder.Event(&df, corev1.EventTypeNormal, "Rollout", "Starting a rollout")
+
+			// Start rollout and update status
+			// update status so that we can track progress
+			df.Status.Phase = PhaseRollout
+			if err := r.Status().Update(ctx, &df); err != nil {
+				log.Error(err, "could not update the Dragonfly object")
+				return ctrl.Result{Requeue: true}, err
+			}
+
+			// requeue so that the rollout is processed
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
why wait 5 secs here?

Reply:
Without a specific period, and just `Requeue: True`, It's taking a while to reconcile the object. 5 seconds is just a good enough timer. Should we go even lower? :thinking:

Reply:
ok though I don't see any action take we need waiting for, in any case IIUC this happens once for every rolling update so not very important.

Reply:
Yep!